src/audio/google_rtc_audio_processing.c

```diff
@@ -0,0 +1,387 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2021 Google LLC. All rights reserved.
+//
+
+#include <errno.h>
+#include <ipc/control.h>
+#include <ipc/stream.h>
+#include <ipc/topology.h>
+#include <google_rtc_audio_processing.h>
+#include <sof/audio/buffer.h>
+#include <sof/audio/component.h>
+#include <sof/audio/format.h>
+#include <sof/audio/kpb.h>
+#include <sof/common.h>
+#include <sof/debug/panic.h>
+#include <sof/ipc/msg.h>
+#include <sof/lib/alloc.h>
+#include <sof/lib/memory.h>
+#include <sof/lib/notifier.h>
+#include <sof/lib/uuid.h>
+#include <sof/lib/wait.h>
+#include <sof/list.h>
+#include <sof/math/numbers.h>
+#include <sof/string.h>
+#include <sof/trace/trace.h>
+#include <sof/ut.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <user/trace.h>
+
+#define GOOGLE_RTC_AUDIO_PROCESSING_SAMPLERATE 48000
+
+/* b780a0a6-269f-466f-b477-23dfa05af758 */
+DECLARE_SOF_RT_UUID("google-rtc-audio-processing", google_rtc_audio_processing_uuid,
+					0xb780a0a6, 0x269f, 0x466f, 0xb4, 0x77, 0x23, 0xdf, 0xa0,
+					0x5a, 0xf7, 0x58);
+DECLARE_TR_CTX(google_rtc_audio_processing_tr, SOF_UUID(google_rtc_audio_processing_uuid),
+			   LOG_LEVEL_INFO);
+
+struct google_rtc_audio_processing_comp_data {
+	struct comp_buffer *raw_microphone;
+	struct comp_buffer *aec_reference;
+	struct comp_buffer *output;
+	uint32_t num_frames;
+	GoogleRtcAudioProcessingState *state;
+	int16_t *aec_reference_buffer;
+	int16_t aec_reference_buffer_index;
+	int16_t *raw_mic_buffer;
+	int16_t raw_mic_buffer_index;
+	int16_t *output_buffer;
+	int16_t output_buffer_index;
+	struct comp_data_blob_handler *tuning_handler;
+};
+
+void *GoogleRtcMalloc(size_t size);
+void GoogleRtcFree(void *ptr);
+
+void *GoogleRtcMalloc(size_t size)
+{
+	return rballoc(0, SOF_MEM_CAPS_RAM, size);
+}
+
+void GoogleRtcFree(void *ptr)
+{
+	return rfree(ptr);
+}
+
+static int google_rtc_audio_processing_params(
+	struct comp_dev *dev,
+	struct sof_ipc_stream_params *params)
```

Comment:
you have a rather unusual c-style, usually we don't finish a line with an open parenthesis, do we?

Reply:
I usually don't either!
I just want to get the line shorter. If I align on the opening parenthesis lines become longer and harder to read.
In that case doing so allow to have a argument list and sorter lines.
If you feel strong against, please suggest your preferred way: I'll be happy to comply.